### PR TITLE
GIVCAMP-289: Inline External Script Loading.

### PR DESCRIPTION
**Pull request by sherakama**
# READY FOR REVIEW

# Summary
- Allows content authors to embed external scripts and markup inline with content
- Provides container controls
- Added new block to content only for now
- TBD permissions to this block type?

# Review By (Date)
End of the sprint

# Criticality
Normal

# Review Tasks

1. Check out this branch locally and fire up an https dev environment
2. Review the editor at https://app.storyblok.com/#/me/spaces/1005200/stories/0/0/4582847
3. Navigate to the embed script block in the content
4. Review fields, descriptions, and options
5. Navigate to the block library and review the 'Embed Script' component (Atoms -> sbEmbedScript) https://app.storyblok.com/#/me/spaces/1005200/components/145576
6. Try out a few of the options and/or load additional scripts
7. Review code

### CODE REVIEW

File: components/Embed/Embed.tsx [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+  className?: string;
+}
+
+const Embed = ({
+id, src, content, boundingWidth, spacingTop, spacingBottom, className, width, ...props
+}:EmbedProps) => {
+
+  const myEmbed = useRef(null);
+
+  useEffect(() => {
```
Comment (sherakama) [inline components/Embed/Embed.tsx:32]: I went with the script element and dom fragment approach as it gave me control over putting these scripts inline with the HTML whereas the next script component did not.

Comment (yvonnetangsu) [reply components/Embed/Embed.tsx:32]: Was just about to ask about the Next Script - guess the old approach is still better huh 👀

File: components/Storyblok/SbEmbed.tsx [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    spacingTop?: PaddingType;
+    spacingBottom?: PaddingType;
+  };
+};
+
+export const SbEmbed = ({ blok, blok: { _uid } }:SbEmbedProps) => {
+
+  return (
+    <Embed
+      {...storyblokEditable}
```
Comment (yvonnetangsu) [inline components/Storyblok/SbEmbed.tsx:22]:
```suggestion
      {...storyblokEditable(blok)}
```
If we pass in `blok` here, you can click on the embedded content in the visual editor to bring you into the component editing sidebar 😄

**Review by yvonnetangsu — APPROVED** [review]
Preapproving with just that one small request in the comment and if you could please resolve the merge conflict. Thank you for this @sherakama ! 🚀 😄

File: components/Embed/Embed.tsx [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+        pt={spacingTop}
+        pb={spacingBottom}
+        className={className}
+      >
+        <div ref={myEmbed} />
+      </WidthBox>
+    );
+  }
+
+  return (<p className='bg-red text-white text-3xl'>You must provide either an src or content to the embed</p>);
```
Comment (yvonnetangsu) [inline components/Embed/Embed.tsx:86]:
```suggestion
  return (<p className='bg-digital-red text-white text-30'>You must provide either an src or content to the embed</p>);
```
Apparently `bg-red` isn't a thing 🤣 Need to attach a number after (the out of the box TW colors). I'm also making the text a bit bigger here.
https://tailwindcss.com/docs/background-color

File: components/Storyblok/SbEmbed.tsx (hunk 2 above)
Comment (sherakama) [reply components/Storyblok/SbEmbed.tsx]: whoops, thanks.

**Comment by yvonnetangsu** [comment]
Thank you for the update @sherakama 🎉  I'll merge this in, then cut a release with both of our PRs

File: components/Embed/Embed.tsx [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+import { type PaddingType } from '@/utilities/datasource';
+
+export interface EmbedProps extends  React.HTMLAttributes<HTMLDivElement> {
+  id?: string;
+  src?: string;
+  content?: string;
+  boundingWidth?: 'site' | 'full';
+  width?: WidthType;
+  spacingTop?: PaddingType;
+  spacingBottom?: PaddingType;
```
Comment (yvonnetangsu) [inline components/Embed/Embed.tsx:22]: Thank you for using these! 😄

**Pull request merged**